src/glove.c

```diff
@@ -64,33 +67,46 @@ int scmp( char *s1, char *s2 ) {
 }
 
 void initialize_parameters() {
+    srand(seed); // set the random seed
     long long a, b;
     vector_size++; // Temporarily increment to allocate space for bias
-
     /* Allocate space for word vectors and context word vectors, and correspodning gradsq */
     a = posix_memalign((void **)&W, 128, 2 * vocab_size * vector_size * sizeof(real)); // Might perform better than malloc
     if (W == NULL) {
         fprintf(stderr, "Error allocating memory for W\n");
         exit(1);
     }
     a = posix_memalign((void **)&gradsq, 128, 2 * vocab_size * vector_size * sizeof(real)); // Might perform better than malloc
-    if (gradsq == NULL) {
+	  if (gradsq == NULL) {
         fprintf(stderr, "Error allocating memory for gradsq\n");
         exit(1);
     }
-    for (b = 0; b < vector_size; b++) {
-        for (a = 0; a < 2 * vocab_size; a++) {
-            W[a * vector_size + b] = (rand() / (real)RAND_MAX - 0.5) / vector_size;
+    // Load existing parameters
+    if (load_init_param) {
+        fprintf(stderr, "\nLoading initial parameters from %s \n", param_file);
+        FILE *fin;
+        fin = fopen(param_file, "rb");
+        if (fin == NULL) {
+          fprintf(stderr, "Unable to open file.\n");
+          exit(1);
         }
-    }
-    for (b = 0; b < vector_size; b++) {
-        for (a = 0; a < 2 * vocab_size; a++) {
-            gradsq[a * vector_size + b] = 1.0; // So initial value of eta is equal to initial learning rate
+        for (a = 0; a < 2 * (long long)vocab_size * (vector_size); a++) {
+            if (feof(fin)) {
+                fprintf(stderr, "EOF reached before parameters fully loaded.\n");
+                exit(1);
+            }
+            fread(&W[a], sizeof(real), 1, fin);
         }
+        fclose(fin);
+    // Initialize new parameters
+    } else {
+        for (b = 0; b < vector_size; b++) for (a = 0; a < 2 * vocab_size; a++) W[a * vector_size + b] = (rand() / (real)RAND_MAX - 0.5) / vector_size;
```

Comment:
Just throwing this thought out there, bunching up lines like this makes it hard to read that diffs